src/mqtt.service.ts

```diff
@@ -3,14 +3,7 @@ import { ISubscriptionGrant } from './mqtt-types';
 import { connect } from '../vendor/mqtt.browserified.js';
 import * as extend from 'xtend';
 
```

Comment:
Isn't importing everything directly from `rxjs` blowing up the size of the bundle, since it's not tree shakeable anymore?

Reply:
No not anymore with the cli v6 (together with webpack 4). Rxjs v6 is marked as sideEffect-free in package.json which allows webpack to remove all the code which is not used. You can read more about it here: https://github.com/webpack/webpack/tree/master/examples/side-effects

The rxjs-compat package only exist to allow a smooth migration to the new api of rxjs and is not tree-shakable. Here are some more information: https://github.com/ReactiveX/rxjs/blob/master/MIGRATION.md